### DISCUSSION TITLE
Better descriptions for hooks config options

### DISCUSSION THREAD

**Discussion by chrisco512** [wevm/wagmi#478] [Ideas]
I think the hook config options could use better descriptions. While working with wagmi, I've been trying to grasp what the effects of all the various config options are and how they interact with each other. For example, if I set `watch: true` does it matter if `cacheTime` or `staleTime` are set? Will it refresh either way? Also, what is the difference between `cacheTime` and `staleTime`? If a value is past its `cacheTime` but not past its `staleTime`, what are the implications? If a value passes its `staleTime` will the framework automatically fetch the latest value? Same question for `cacheTime`?

**Comment by tmm** [comment]
Definitely! In the meantime, you can check out how React Query [explains `staleTime` versus `cacheTime`](https://react-query.tanstack.com/guides/caching). wagmi uses React Query's `QueryObserver` behind the scenes so a lot of the concepts are the same.
